e2e/tests/node_not_ready/node_power_off.robot

```diff
@@ -0,0 +1,491 @@
+*** Settings ***
+Documentation       Test the Longhorn resillence if cluster node powering off
+
+Resource            ../../keywords/common.resource
+Resource            ../../keywords/engine.resource
+Resource            ../../keywords/replica.resource
+Resource            ../../keywords/node.resource
+Resource            ../../keywords/volume.resource
+
+Suite Setup         set_test_suite_environment
+Test Setup          set_test_environment    ${TEST NAME}
+Test Teardown       Cleanup resource and resume state
+
+
+*** Variables ***
+${Gi}=                  2**30
+${volume_size_gb}=      1
+${sleep_interval}=      300
+${volume_type}=         RWO
+
+
+*** Test Cases ***
+Node power off with replica-VA
+    ${number_of_replicas}=    Convert To Integer    3
+    Set Test Variable    ${number_of_replicas}
+
+    ${volume_type}=    Evaluate    "${volume_type}".lower()
+
+    ${field1}=    Convert To String    {"spec": {"size": "${${volume_size_gb} * ${Gi}}"}}
+    ${field2}=    Convert To String    {"spec": {"numberOfReplicas": ${number_of_replicas}}}
+    ${field3}=    Convert To String    {"spec": {"accessMode": "${volume_type}"}}
+    @{list_of_fields}=    Create List    ${field1}    ${field2}    ${field3}
+
+    Given Create Volume With Fields    ${list_of_fields}
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 1
+
+    Then Node 1 should have 1 volume replica
+    And Node 1 state should eventually be NotReady
+    And Volume state should eventually be unknown
+    And Engine state should eventually be unknown
+    And Replica state on node 1 should eventually be unknown
+
+    When Power on node 1
+
+    Then Node 1 state should eventually be Ready
+    And Volume state should eventually be healthy
+    And Engine state should eventually be running
+    And All replicas state should eventually be running
+    And Data should be intact
+    [Teardown]    Teardown
+
+Node power off with replica
+    ${number_of_replicas}=    Convert To Integer    3
+    Set Test Variable    ${number_of_replicas}
+
+    ${volume_type}=    Evaluate    "${volume_type}".lower()
+
+    ${field1}=    Convert To String    {"spec": {"size": "${${volume_size_gb} * ${Gi}}"}}
+    ${field2}=    Convert To String    {"spec": {"numberOfReplicas": ${number_of_replicas}}}
+    ${field3}=    Convert To String    {"spec": {"accessMode": "${volume_type}"}}
+    @{list_of_fields}=    Create List    ${field1}    ${field2}    ${field3}
+
+    Given Create Volume With Fields    ${list_of_fields}
+    And Attach volume to node 1
+    And Write data into mount point
+
+    When Power off node 2
+
+    Then Node 1 should have 1 volume replica
+    And Node 2 should have 1 volume replica
+    And Node 2 state should eventually be NotReady
+    And Volume state should eventually be degraded
+    And Engine state should eventually be running
```

Comment:
Is there any state change here? The engine state should already be running, right? If true, then we should use `Engine state should be <...>` instead.

Reply:
Yes, I've checked all cases and updated the description that the workload hasn't changed to `<workload> state should be <state>`.
https://github.com/longhorn/longhorn-tests/blob/5290967ed6110b1c7dbb194db19be29810102519/e2e/tests/node_not_ready/node_power_off.robot#L253-L258